linux_os/guide/services/sssd/sssd_enable_pam_services/oval/shared.xml

```diff
@@ -2,28 +2,40 @@
   <definition class="compliance" id="sssd_enable_pam_services" version="1">
     <metadata>
       <title>Configure PAM in SSSD Services</title>
-      <affected family="unix">
-        <platform>Red Hat Enterprise Linux 7</platform>
-        <platform>Red Hat Enterprise Linux 8</platform>
-        <platform>Red Hat Virtualization 4</platform>
-        <platform>multi_platform_ol</platform>
-      </affected>
+      {{{- oval_affected(products) }}}
       <description>SSSD should be configured to run SSSD PAM services.
       </description>
     </metadata>
     <criteria>
+      {{% macro check_sssd_pam_criterion(type) %}}
       <criterion comment="check if pam is configured in the services setting of the sssd section"
-      test_ref="test_sssd_enable_pam_services" />
+      test_ref="test_sssd_enable_pam_services_{{{ type }}}" />
+      {{% endmacro %}}
+
+      {{{ check_sssd_pam_criterion("file") }}}
+      {{{ check_sssd_pam_criterion("dir") }}}
     </criteria>
   </definition>
+
+  {{% macro check_sssd_pam_test(type, path, file_pattern) %}}
   <ind:textfilecontent54_test check="all" check_existence="all_exist"
   comment="check if pam is configured in the services setting of the sssd section"
-  id="test_sssd_enable_pam_services" version="1">
-    <ind:object object_ref="obj_sssd_enable_pam_services" />
+  id="test_sssd_enable_pam_services_{{{ type }}}" version="1">
+    <ind:object object_ref="obj_sssd_enable_pam_services_{{{ type }}}" />
   </ind:textfilecontent54_test>
-  <ind:textfilecontent54_object id="obj_sssd_enable_pam_services" version="1">
-    <ind:filepath>/etc/sssd/sssd.conf</ind:filepath>
+  <ind:textfilecontent54_object id="obj_sssd_enable_pam_services_{{{ type }}}" version="1">
+    {{% if type == "file" %}}
+    <ind:filepath>{{{ path }}}</ind:filepath>
+    {{% else %}}
+    <ind:path>{{{ path }}}</ind:path>
+    <ind:filename operation="pattern match">{{{ file_pattern }}}</ind:filename>
+    {{% endif %}}
     <ind:pattern operation="pattern match">^[\s]*\[sssd]([^\n\[\]]*\n+)+?[\s]*services.*pam.*$</ind:pattern>
     <ind:instance datatype="int">1</ind:instance>
   </ind:textfilecontent54_object>
+  {{% endmacro %}}
+
+  {{{ check_sssd_pam_test("file", "/etc/sssd/sssd.conf") }}}
+  {{{ check_sssd_pam_test("dir", "/etc/sssd/conf.d/", "^.*\.conf$") }}}
+
```

Comment:
I don't think we need all these checks. A simple regex in `<ind:filepath>` will do.

Reply:
Done. Thanks.